### DISCUSSION TITLE
About collision query implementation

### DISCUSSION THREAD

**Discussion by Ivan-le-sketch** [amerkoleci/JoltPhysicsSharp#110] [Q&A]
Sorry if my question sounds dumb, I have next to no knowledge when it comes to language interop.

If I'm understanding things right, BroadPhaseQuery and NarrowPhaseQuery methods all take a function pointer to a callback method which is used to create a CollisionCollector object. Is there a reason we can't have a handle to manipulate CollisionCollector objects and use this in the collision query api instead ?

**Comment by amerkoleci** [comment]
Hi, 
Just simplicity, collectors in jolt are generic/template classes and not sure how I can represent them in C world, ideas are welcome 😀

**Comment by Ivan-le-sketch** [comment]
Isn't creating explicitly typed bindings for every combination of type parameters technically possible ?

This is just a question though. I don't have an actual problem this would solve.
